annotation-processor/src/main/java/org/hibernate/validator/ap/checks/ConstraintCheckFactory.java

```diff
@@ -55,7 +60,15 @@ public ConstraintCheckFactory(Types typeUtils, ConstraintHelper constraintHelper
 		fieldChecks = CollectionHelper.newHashMap();
 		fieldChecks.put(
 				AnnotationType.CONSTRAINT_ANNOTATION,
-				new SingleValuedChecks( new StaticCheck(), new TypeCheck( constraintHelper ) )
+				new SingleValuedChecks(
+						new StaticCheck(),
+						new TypeCheck( constraintHelper ),
+						new AnnotationParametersSizeLengthCheck( annotationApiHelper ),
+						new AnnotationParametersPatternCheck( annotationApiHelper ),
+						new AnnotationParametersScriptAssertCheck( annotationApiHelper ),
+						new AnnotationParametersDigitsCheck( annotationApiHelper ),
+						new AnnotationParametersDecimalMinMaxCheck( annotationApiHelper )
+				)
 		);
 		fieldChecks.put(
 				AnnotationType.MULTI_VALUED_CONSTRAINT_ANNOTATION,
```

Comment:
I've added the cases for this kind of constraints and it didn't work out out of the box - so I've added those checks in MULTI_VALUED_CONSTRAINT_ANNOTATION as well.